Sources/MapboxMaps/MapView/Configuration/MapInitOptions.swift

```diff
@@ -0,0 +1,35 @@
+import Foundation
+
+@objc public protocol MapInitOptionsDataSource {
+    /// When you implement this method you should return a `MapInitOptions`.
+    func mapInitOptions() -> Any
+}
+
+/// Options used when initializing `MapView`.
+///
+/// Contains the `ResourceOptions`, `MapOptions` (including `GlyphsRasterizationOptions`)
+/// that are required to initialize a `MapView`.
+public struct MapInitOptions: Equatable {
```

Comment:
What if we just made this a `class` so that it works with the `@objc` protocol? It's immutable, so it should be fine. We'd just need to implement `==`.

Reply:
It's not sufficient to be a `class` - it would need to be an `NSObject`. Are we good with that?

Reply:
I am. I think the type-safety outweighs the avoidance of NSObject.